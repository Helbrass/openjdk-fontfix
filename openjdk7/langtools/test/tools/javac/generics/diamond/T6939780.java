/*
 * @test /nodynamiccopyright/
 * @bug 6939780 7020044
 *
 * @summary  add a warning to detect diamond sites
 * @author mcimadamore
 * @compile/ref=T6939780.out T6939780.java -XDrawDiagnostics -XDfindDiamond
 *
 */

class T6939780 {

    void test() {
        class Foo<X extends Number> {
            Foo() {}
            Foo(X x) {}
        }
        Foo<Number> f1 = new Foo<Number>(1);
        Foo<?> f2 = new Foo<Number>();
        Foo<?> f3 = new Foo<Integer>();
        Foo<Number> f4 = new Foo<Number>(1) {};
        Foo<?> f5 = new Foo<Number>() {};
        Foo<?> f6 = new Foo<Integer>() {};
    }
}
