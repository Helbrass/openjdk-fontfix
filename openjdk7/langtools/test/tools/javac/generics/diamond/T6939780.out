T6939780.java:19:28: compiler.warn.diamond.redundant.args: Foo<java.lang.Number>, Foo<java.lang.Number>
T6939780.java:20:28: compiler.warn.diamond.redundant.args.1: Foo<java.lang.Integer>, Foo<java.lang.Number>
2 warnings
